node/pkg/telemetry/prom_remote_write/prom_test.go

```diff
@@ -0,0 +1,194 @@
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "rpc_duration_seconds"},
+				{Name: "job", Value: "promtool"},
+				{Name: "quantile", Value: "0.5"},
+			},
+			Samples: []prompb.Sample{{Value: 4773, Timestamp: 1}},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "rpc_duration_seconds"},
+				{Name: "job", Value: "promtool"},
+				{Name: "quantile", Value: "0.99"},
+			},
+			Samples: []prompb.Sample{{Value: 76656, Timestamp: 1}},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "rpc_duration_seconds_sum"},
+				{Name: "job", Value: "promtool"},
+			},
+			Samples: []prompb.Sample{{Value: 1.7560473e+07, Timestamp: 1}},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "rpc_duration_seconds_count"},
+				{Name: "job", Value: "promtool"},
+			},
+			Samples: []prompb.Sample{{Value: 2693, Timestamp: 1}},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "test_metric1"},
+				{Name: "b", Value: "c"},
+				{Name: "baz", Value: "qux"},
+				{Name: "d", Value: "e"},
+				{Name: "foo", Value: "bar"},
+				{Name: "job", Value: "promtool"},
+			},
+			Samples: []prompb.Sample{{Value: 1, Timestamp: 1}},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: "__name__", Value: "test_metric1"},
+				{Name: "b", Value: "c"},
+				{Name: "baz", Value: "qux"},
+				{Name: "d", Value: "e"},
+				{Name: "foo", Value: "bar"},
+				{Name: "job", Value: "promtool"},
+			},
+			Samples: []prompb.Sample{{Value: 2, Timestamp: 1}},
+		},
+	},
+}
+
+func TestParseAndPushMetricsTextAndFormat(t *testing.T) {
+	input := bytes.NewReader([]byte(`
+	# HELP http_request_duration_seconds A histogram of the request duration.
+	# TYPE http_request_duration_seconds histogram
+	http_request_duration_seconds_bucket{le="0.1"} 33444 1
+	http_request_duration_seconds_bucket{le="0.5"} 129389 1
+	http_request_duration_seconds_bucket{le="1"} 133988 1
+	http_request_duration_seconds_bucket{le="+Inf"} 144320 1
+	http_request_duration_seconds_sum 53423 1
+	http_request_duration_seconds_count 144320 1
+	# HELP http_requests_total The total number of HTTP requests.
+	# TYPE http_requests_total counter
+	http_requests_total{method="post",code="200"} 1027 1395066363000
+	http_requests_total{method="post",code="400"}    3 1395066363000
+	# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+	# TYPE rpc_duration_seconds summary
+	rpc_duration_seconds{quantile="0.01"} 3102 1
+	rpc_duration_seconds{quantile="0.5"} 4773 1
+	rpc_duration_seconds{quantile="0.99"} 76656 1
+	rpc_duration_seconds_sum 1.7560473e+07 1
+	rpc_duration_seconds_count 2693 1
+	# HELP test_metric1 This is a test metric.
+	# TYPE test_metric1 gauge
+	test_metric1{b="c",baz="qux",d="e",foo="bar"} 1 1
+	test_metric1{b="c",baz="qux",d="e",foo="bar"} 2 1
+	`))
+	labels := map[string]string{"job": "promtool"}
+
+	expected, err := MetricTextToWriteRequest(input, labels)
```

Comment:
Extreme nit, but should "expected" be "actual"?

Reply:
Changed